### PR TITLE
feat: support dark theme

### PR DESCRIPTION
**Pull request by georgesamy**
Support dark theme for the GUI. It uses chakra-ui's built in **Color Mode** and Monaco Editor's **theme**

Note: the `<SunIcon />` was used in the Donate button. I wanted to use it in the theme button so I reassigned it to `<BellIcon />`. Not a hard requirement but I thought it would be nicer. Open to changes.

### CODE REVIEW

**Comment by glassmonkey** [comment]
Thank you for your Pull Request. I appreciate your effort in contributing to the project.
 I'll review it shortly and provide feedback or any necessary follow-up.

**Comment by georgesamy** [comment]
Ofcourse! and thanks for creating this project, been using it for a while

**Comment by glassmonkey** [comment]
@georgesamy 

Thank you your PullRequest.
I wanted to express my sincere gratitude for the feature you added. 
It has significantly improved the usability of this project, making it much more comfortable to work with.

I have one small suggestion to further enhance the user experience.
 Would it be possible to replace the current button with a toggle switch? 

This change would make it clearer whether the button shows the current state or changes the state upon being pressed. I believe the following Pull Request might serve as a good reference for this enhancement: https://github.com/chakra-ui/chakra-ui/issues/3518.

https://github.com/chakra-ui/chakra-ui/issues/3518

File: src/app.tsx [hunk 1]
```diff
@@ -124,7 +127,15 @@ export default function App() {
...
-							leftIcon={<SunIcon />}
+							leftIcon={ ( colorMode === "light" ) ? <MoonIcon /> : <SunIcon /> }
+							onClick={toggleColorMode}
+							colorScheme="blue"
+							mr={2}
+						>
+							{ (colorMode === "light") ? "Dark" : "Light" }
+						</Button>
+						<Button
+							leftIcon={<BellIcon />}
```
Comment (glassmonkey) [inline src/app.tsx:138]: nice idea!!

**Comment by georgesamy** [comment]
I looked into the issue you posted and there wasn't an obvious solution.  You see I'm not good on the frontend side xD
I'll try to give this another look over the next week

**Comment by glassmonkey** [comment]
thank you

**Comment by georgesamy** [comment]
Ok it's done.. Also moved the switch to the top right corner.

**Comment by glassmonkey** [comment]
@georgesamy 

Thank you for your Pull Request! I've successfully merged it into the project. Additionally, I made some further position adjustments on our end to ensure optimal integration. Your contribution is greatly appreciated!

https://github.com/glassmonkey/php-playground/pull/134

**Pull request merged**